msal/oauth2cli/oidc.py

```diff
@@ -184,6 +184,8 @@ def initiate_auth_code_flow(
             # Here we just automatically add it. If the caller do not want id_token,
             # they should simply go with oauth2.Client.
             _scope.append("openid")
+        if kwargs.get("response_mode") not in (None, "form_post", "query"):
+            raise ValueError('only response_mode="form_post" or response_mode="query" is allowed. default is query')
```

Comment:
The `oidc.py` is designed to be a generic and stable OIDC layer which already accepts all (and even future) `response_mode` values. There is no need to add this check here. If you really want to have such check, please move this part to `application.py` line 712.5. Otherwise, we can just remove them here.

```suggestion
```